app/src/main/java/com/hedvig/app/feature/profile/ui/payment/PaymentHistoryActivity.kt

```diff
@@ -19,16 +22,22 @@ class PaymentHistoryActivity : BaseActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_payment_history)
 
-        setupLargeTitle(
-            R.string.PAYMENT_HISTORY_TITLE,
-            R.drawable.ic_back,
-            backAction = {
-                onBackPressed()
-            })
+        paymentHistory.setEdgeToEdgeSystemUiFlags(true)
+        paymentHistory.doOnApplyWindowInsets { view, insets, initialState ->
+            view.updatePadding(bottom = initialState.paddings.bottom + insets.systemWindowInsetBottom)
+        }
+        toolbar.doOnApplyWindowInsets { view, insets, initialState ->
+            view.updatePadding(top = initialState.paddings.top + insets.systemWindowInsetTop)
+        }
+        toolbar.setNavigationOnClickListener {
+            onBackPressed()
+        }
+        paymentHistory.setupToolbarScrollListener(toolbar)
 
         profileViewModel.data.observe(lifecycleOwner = this) { data ->
             data?.chargeHistory?.let { chargeHistory ->
-                paymentHistory.adapter = PaymentHistoryAdapter(wrapCharges(chargeHistory))
+                paymentHistory.adapter =
+                    PaymentHistoryAdapter(listOf(ChargeWrapper.Title) + wrapCharges(chargeHistory))
```

Comment:
There is some room for improvement here if you feel like it: We could make the `PaymentHistoryAdapter` not take its data in the constructor, in a manner similar to our more recent `RecyclerView.Adapter`s.

Not extremely necessary but it would be neat I suppose